config/config.go

```diff
@@ -0,0 +1,10 @@
+package config
+
+import "github.com/spf13/viper"
+
+func LoadConfig(configPath string) error {
+	viper.SetConfigName("conf")
+	viper.SetConfigType("yaml")
+	viper.AddConfigPath(configPath)
+	return viper.ReadInConfig() // Find and read the config file
```

Comment:
looks like the values from the conf file are not respected (ie, got overridden with the default flag's value) - `timeout` example:

```
$ cat resources/config/conf.yaml
plugins:
  trivy:
    exit-code: 0
    timeout: 51m0s
    severity: HIGH,CRITICAL
    ignore-unfixed: true
    security-checks: vuln,config
    skip-files: ""
    skip-dirs: ""
    image-ref: ""
    version: ""
    helm-overrides-files: ""
```

```
$ go run main.go trivy
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          timeout : 5m0s
          severity: HIGH,CRITICAL
          ignore-unfixed: true
          security-checks: vuln,config
```

additionally, how would you detect if a key/parameter is missing - ie, was not provided via the conf file and does it gets assigned the default value based on type in that case?

this is important eg, as we might expand the config scheme with new params tools gets with time, but the old configs would not have them

Reply:
If no config file is provided then we are taking default config file path to ```conf.yaml```.If for any reason the config file path is wrong or ```plugins.trivy'''  key is missing from the yaml then we are considering the command line defaults from the flags.
If the ```conf.yaml``` is
```
plugins:
  trivy:
    exit-code: 0
    timeout: 51m0s
    severity: HIGH
    ignore-unfixed: true
    security-checks: vuln,config
    skip-files: ""
    skip-dirs: ""
    image-ref: ""
    version: ""
    helm-overrides-files: ""
```
and if you don't specify the config file or if you correct config file in ```---config``` then values from the configuration would be used.
```
$ go run main.go trivy --config=conf.yaml
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          timeout : 51m0s
          severity: HIGH
          ignore-unfixed: true
          security-checks: vuln,config                                                                                                                             
$ dynamic-buildkite-template % go run main.go trivy                   
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          timeout : 51m0s
          severity: HIGH
          ignore-unfixed: true
          security-checks: vuln,config     
          ```
          However you can still override it using command line flags
          ```
          $ go run main.go trivy --timeout=6m43s   
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          timeout : 6m43s
          severity: HIGH,CRITICAL
          ignore-unfixed: true
          security-checks: vuln,config                                                                                                                            
$ go run main.go trivy --config=conf.yaml --timeout=6m43s
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          timeout : 6m43s
          severity: HIGH,CRITICAL
          ignore-unfixed: true
          security-checks: vuln,config       
          ```
          If wrong file is provided then defaults from the command line flags would be used
          ```
         $ go run main.go trivy --config=x.yaml                   
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          timeout : 5m0s
          severity: HIGH,CRITICAL
          ignore-unfixed: true
          security-checks: vuln,config%         
          ```
          and here also you can still override the defaults by passing values in command line flags
```
$ go run main.go trivy --config=x.yaml --timeout=6m43s
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          timeout : 6m43s
          severity: HIGH,CRITICAL
          ignore-unfixed: true
          security-checks: vuln,config                       
```

Reply:
Hi @prezha I have created a separate ticket for this fix and am currently working on it. 
The reference ticket as follows :[JIRA ticket](https://packet.atlassian.net/browse/SCT-439)

Reply:
@hnadiminti-equinix is there any specific reason why we want to check every flag in the template?
removing that might be a workaround for skipping zero values - eg:

with templates/trivy-step.tmpl:
```
steps:
  - command: ls
    plugins:
{{- if .TrivyPluginEnabled }}
      - equinixmetal-buildkite/trivy#{{ .TPConfig.TrivyVersion }}:
          exit-code: {{ .TPConfig.ExitCode }}
          timeout : {{ .TPConfig.Timeout }}
          severity: {{ .TPConfig.Severity }}
          ignore-unfixed: {{ .TPConfig.IgnoreUnfixed }}
          security-checks: {{ .TPConfig.SecurityChecks }}
          skip-files: '{{ .TPConfig.SkipFiles }}'
          skip-dirs: '{{ .TPConfig.SkipDirs }}'
          image-ref: '{{ .TPConfig.ImageRef }}'
          helm-overrides-file: '{{ .TPConfig.HelmOverridesFiles }}'
{{- end}}
```
default config would be:
```
$ go run . trivy --ignore-unfixed=false
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          exit-code: 0
          timeout : 5m0s
          severity: HIGH,CRITICAL
          ignore-unfixed: false
          security-checks: vuln,config
          skip-files: ''
          skip-dirs: ''
          image-ref: ''
          helm-overrides-file: '
```
and with override flags:
```
$ go run . trivy --ignore-unfixed=true --exit-code=42
steps:
  - command: ls
    plugins:
      - equinixmetal-buildkite/trivy#v1.18.2:
          exit-code: 42
          timeout : 5m0s
          severity: HIGH,CRITICAL
          ignore-unfixed: true
          security-checks: vuln,config
          skip-files: ''
          skip-dirs: ''
          image-ref: ''
          helm-overrides-file: '
```
btw, not sure why last `'` is missing in the output